src/ScenarioStateArgumentOrganiser.php

```diff
@@ -34,21 +37,28 @@ public function __construct(ArgumentOrganiser $organiser, ScenarioStateInitializ
      */
     public function organiseArguments(ReflectionFunctionAbstract $function, array $match)
     {
-        $store = $this->store->getStore();
-
-        //@todo, be more defensive
         $i = array_slice(array_keys($match), -1, 1)[0];
-
-        $parameters = $function->getParameters();
-
         $paramsKeys = array_map(function ($element) {
             return $element->name;
-        }, $parameters);
+        }, $function->getParameters());
+
+        $store = $this->store->getStore();
+        $reader = new AnnotationReader();
+
+        // Ignore Behat annotations
+        $reader::addGlobalIgnoredName('Given');
+        $reader::addGlobalIgnoredName('When');
+        $reader::addGlobalIgnoredName('Then');
 
-        foreach ($paramsKeys as $key) {
-            if ($store->hasStateFragment($key)) {
-                $match[$key] = $store->getStateFragment($key);
-                $match[strval(++$i)] = $store->getStateFragment($key);
+        if (null !== ($annotation = $reader->getMethodAnnotation($function, ScenarioStateArgument::class))) {
```

Comment:
Better use [getMethodAnnotations](https://github.com/doctrine/annotations/blob/master/lib/Doctrine/Common/Annotations/AnnotationReader.php#L251) instead, to match multiples instances of same annotation ?